src/utils/tezos.util.ts

```diff
@@ -16,11 +16,19 @@ export const RPC_RETRY_OPTIONS = {
 };
 
 export const mutezToTz = (x: BigNumber, decimals: number) => {
-  return new BigNumber(x).integerValue().shiftedBy(-decimals);
+  if (x.isNaN()) {
+    return x;
+  }
+
+  return x.integerValue().div(new BigNumber(10).pow(decimals));
```

Comment:
as for me, the previous versions with `shiftedBy` were more convenient, just add the `isNaN` check